nestkernel/parameter.cpp

```diff
@@ -99,9 +100,11 @@ NormalParameter::NormalParameter( const DictionaryDatum& d )
 }
 
 double
-NormalParameter::value( RngPtr rng, Node* )
+NormalParameter::value( RngPtr rng, Node* node )
 {
-  return normal_dists_[ kernel().vp_manager.get_thread_id() ]( rng );
+  const auto vp = kernel().vp_manager.node_id_to_vp( node->get_node_id() );
+  const auto tid = kernel().vp_manager.vp_to_thread( vp );
```

Comment:
This makes sense, but I am a bit worried about performance and code duplication. We do exactly the same operations here:

https://github.com/nest/nest-simulator/blob/062e2723e1df19433e4c0131f1aa36a428067156/libnestutil/dict_util.h#L56-L57

right before calling this method. Both steps involve modulus or division, which are expensive. Have you checked the effect on perfomance? How about passing the `tid` from the calling function? Or would that cause problems elsewhere? Also, in `dict_util.h`, we could add `const` qualifiers which may allow more aggressive optimisation of chained modulus and division.